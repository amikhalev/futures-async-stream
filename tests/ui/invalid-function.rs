// compile-fail

#![deny(warnings)]
#![feature(async_await, generators)]

use futures_async_stream::async_stream;

#[async_stream(item = ())]
const fn constness() {} //~ ERROR async stream may not be const

#[async_stream(item = ())]
fn variadic(_: ...) {} //~ ERROR async stream may not be variadic

#[async_stream(item = ())]
fn asyncness() {} //~ ERROR async stream must be declared as async

#[async_stream(item = ())]
async fn output() -> i32 {} //~ ERROR async stream must return the unit type

#[async_stream(item = ())]
async fn unit() -> () {} // OK

fn main() {}
