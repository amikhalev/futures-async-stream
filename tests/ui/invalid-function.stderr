error: only foreign functions are allowed to be C-variadic
  --> $DIR/invalid-function.rs:12:16
   |
12 | fn variadic(_: ...) {} //~ ERROR async stream may not be variadic
   |                ^^^

error: async stream may not be const
 --> $DIR/invalid-function.rs:9:1
  |
9 | const fn constness() {} //~ ERROR async stream may not be const
  | ^^^^^

error: expected `:`
  --> $DIR/invalid-function.rs:11:1
   |
11 | #[async_stream(item = ())]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^

error: async stream must be declared as async
  --> $DIR/invalid-function.rs:15:1
   |
15 | fn asyncness() {} //~ ERROR async stream must be declared as async
   | ^^

error: async stream must return the unit type
  --> $DIR/invalid-function.rs:18:22
   |
18 | async fn output() -> i32 {} //~ ERROR async stream must return the unit type
   |                      ^^^

error: aborting due to 5 previous errors

